tests/backup/backup_rbac_test.go

```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
+		backupLocationMap          = make(map[string]string)
+		periodicSchedulePolicyName string
+		periodicSchedulePolicyUid  string
+		preRuleName                string
+		postRuleName               string
+		preRuleUid                 string
+		postRuleUid                string
+		srcClusterUid              string
+		scheduledBackupName        string
+		userScheduleName           string
+		restoreName                string
+	)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("VerfiyRBACforAppUser", "To verify all the RBAC operations for an app-user", nil, 87889)
+		log.InfoD("scheduling applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, appCtx := range appContexts {
+				appCtx.ReadinessTimeout = appReadinessTimeout
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+				namespace := GetAppNamespace(appCtx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+
+	It("To verify all the RBAC operations for an app-user", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Create an app-user", func() {
+			log.InfoD("Creating a new user and assigning the role of app-user")
+			appUser = createUsers(1)[0]
+			err := backup.AddRoleToUser(appUser, backup.ApplicationUser, fmt.Sprintf("Adding Application User role to %s", appUser))
+			log.FailOnError(err, "Failed to add role to user - [%s]", appUser)
+			userNames = append(userNames, appUser)
+		})
+
+		// Verifying the backup objects creation for an app-user
+		Step(fmt.Sprintf("Verify if the App-User doesn't have permission to create cloud credentials and backup location"), func() {
+			log.InfoD("Verify if the App-User doesn't have permission to create cloud credentials and backup location")
+			ctxNonAdmin, err := backup.GetNonAdminCtx(appUser, commonPassword)
+			log.FailOnError(err, "Fetching non admin ctx")
+			for _, provider := range providers {
+				if provider != drivers.ProviderNfs {
+					cloudCredUID = uuid.New()
+					backupLocationUID = uuid.New()
+					credName = fmt.Sprintf("cred-%s-%v", provider, RandomString(6))
+					err = CreateCloudCredential(provider, credName, cloudCredUID, orgID, ctxNonAdmin)
+					dash.VerifyFatal(strings.Contains(err.Error(), "PermissionDenied"), true, fmt.Sprintf("Verifying if App-User [%s] doesn't have permission for creating cloud credentials for provider [%s]", appUser, provider))
+				} else {
+					log.InfoD("Provider NFS does not require creating cloud credential")
+				}
+				backupLocationName = fmt.Sprintf("backup-location-%s-%v", provider, RandomString(4))
+				err = CreateBackupLocationWithContext(provider, backupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "", ctxNonAdmin)
```

Comment:
Is cloudCredUID initiallzed?

Reply:
I have now changed the block as below;
![image](https://github.com/portworx/torpedo/assets/148203162/8c28021f-2e6a-48da-a2f0-a9a4edc1070d)

Reply:
Modified the code and removed the cloudcredUID from the if block.
![image](https://github.com/portworx/torpedo/assets/105486766/f68ce3a4-d320-46ff-8294-eadc2d9d1ffe)